frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js

```diff
@@ -223,9 +217,36 @@ const getDefaultCustomAnalysisSettings = () => {
   return customVals;
 };
 
-const getArraysForWells = () => {
+const getDefaultAnalysisSettingsForWell = () => {
```

Comment:
What is the difference between this and `getDefaultCustomAnalysisSettings`? This just looks like a redefinition

Reply:
this is for one well that is required right on initialization whereas `getDefaultCustomAnalysisSettings` gets calls once the data has been fetched and the number of wells has been determined

Reply:
and `customAnalysisSettingsForWell` can't be set until the first well is determined, can't assume it's A1. 
```
useEffect(() => {
    if (selectedWell in customAnalysisSettings)
      setCustomAnalysisSettingsForWell(
        // Tanner (6/1/23): Copying just to be safe
        deepCopy({
          windowedAnalysisBounds: customAnalysisSettings.windowedAnalysisBounds,
          featureIndices: customAnalysisSettings[selectedWell].filteredFeatureIndices,
          duplicateIndices: customAnalysisSettings[selectedWell].duplicateFeatureIndices,
          thresholdEndpoints: customAnalysisSettings[selectedWell].thresholdEndpoints,
        })
      );
  }, [customAnalysisSettings, selectedWell]);
```